gene2phenotype_project/gene2phenotype_app/views.py

```diff
@@ -850,3 +851,51 @@ def list(self, request, *args, **kwargs):
                 'data': curation_data_obj.json_data,
             }
         return Response(response_data)
+
+
+class UpdateCurationData(generics.UpdateAPIView):
+    """
+        Updates the JSON data for the specific G2P ID.
+    """
+
+    http_method_names = ['put', 'options']
+    serializer_class = CurationDataSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_queryset(self):
+        stable_id = self.kwargs['stable_id']
+        user = self.request.user
+
+        g2p_stable_id = get_object_or_404(G2PStableID, stable_id=stable_id)
+        # Get the entry for this user
+        queryset = CurationData.objects.filter(stable_id=g2p_stable_id, user__email=user)
+
+        if not queryset.exists():
+            self.handle_no_permission('Entry', stable_id)
+        else:
+            return queryset
+
+    def update(self, request, *args, **kwargs):
+        curation_obj = self.get_queryset().first()
+
+        json_file_path = settings.BASE_DIR.joinpath("gene2phenotype_app", "utils", "curation_schema.json")
+        try:
+            with open(json_file_path, 'r') as file:
+                schema = json.load(file)
+        except FileNotFoundError:
+            return Response({"message": "Schema file not found"}, status=status.HTTP_404_NOT_FOUND)
+        # Validate the JSON data against the schema
+        try:
+            print(request.data)
+            validate(instance=request.data, schema=schema)
+        except exceptions.ValidationError as e:
+            return Response({"message": "JSON data does not follow the required format, Required format is" + str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+        serializer = CurationDataSerializer(curation_obj, data=request.data, context={'user': self.request.user})
+
+        if serializer.is_valid():
+            serializer.save()
```

Comment:
Suggestion - I think this should be `serializer.update(curation_obj, request.data)`, serializer.save should be for create, I mean both of them work but this is the update bit so should use the update function of the serializer

Reply:
Actually considering [this ](https://www.django-rest-framework.org/api-guide/serializers/#saving-instances:~:text=Calling%20.save()%20will%20either%20create%20a%20new%20instance%2C%20or%20update%20an%20existing%20instance%2C%20depending%20on%20if%20an%20existing%20instance%20was%20passed%20when%20instantiating%20the%20serializer%20class%3A), please ignore